adapters/Vungle/VungleAdapter/Bidding/GADMediationVungleBanner.m

```diff
@@ -101,154 +98,130 @@ - (void)requestBannerAd {
+    NSValueFromGADAdSize(shortBannerSize)
+  ];
+
+  GADAdSize closestSize = GADClosestValidSizeForAdSizes(adSize, potentials);
+  CGSize size = CGSizeFromGADAdSize(closestSize);
+  if (size.height == GADAdSizeBanner.size.height) {
+    if (size.width < GADAdSizeBanner.size.width) {
+      return shortBannerSize;
+    } else {
+      return GADAdSizeBanner;
+    }
+  } else if (size.height == GADAdSizeLeaderboard.size.height) {
+    return GADAdSizeLeaderboard;
   }
-  _didBannerFinishPresenting = YES;
-
-  [VungleSDK.sharedSDK finishDisplayingAd:self.desiredPlacement adMarkup:self.bidResponse];
-  [[GADMAdapterVungleBiddingRouter sharedInstance] removeDelegate:self];
-  _bannerView = nil;
-}
-
-#pragma mark - GADMAdapterVungleDelegate delegates
-
-- (NSString *)bidResponse {
-  return [_adConfiguration bidResponse];
-}
-
-- (GADAdSize)bannerAdSize {
-  return _bannerSize;
+  return GADAdSizeInvalid;
 }
 
-- (void)initialized:(BOOL)isSuccess error:(nullable NSError *)error {
-  if (!isSuccess) {
-    _adLoadCompletionHandler(nil, error);
-    return;
+- (BannerSize)convertGADAdSizeToBannerSize {
+  if (GADAdSizeEqualToSize(_bannerSize, GADAdSizeMediumRectangle)) {
+    return BannerSizeMrec;
   }
-  [self loadAd];
+  if (_bannerSize.size.height == GADAdSizeLeaderboard.size.height) {
+    return BannerSizeLeaderboard;
+  }
+  // Height is 50.
+  if (_bannerSize.size.width < GADAdSizeBanner.size.width) {
+    return BannerSizeShort;
+  }
+  return BannerSizeRegular;
 }
 
-- (void)adAvailable {
-  if (_isAdLoaded) {
-    // Already invoked an ad load callback.
-    return;
-  }
-  _isAdLoaded = YES;
+#pragma mark - VungleBannerDelegate
+
+- (void)bannerAdDidLoad:(VungleBanner *)banner {
   _bannerView = [[UIView alloc]
       initWithFrame:CGRectMake(0, 0, _bannerSize.size.width, _bannerSize.size.height)];
-  self.bannerState = BannerRouterDelegateStateWillPlay;
-  NSError *error = [self renderAd];
-  if (error) {
-    _adLoadCompletionHandler(nil, error);
-    return;
-  }
-
   if (_adLoadCompletionHandler) {
+    [_bannerAd presentOn:_bannerView];
     _delegate = _adLoadCompletionHandler(self, nil);
   }
+}
 
-  if (!_delegate) {
-    [[GADMAdapterVungleBiddingRouter sharedInstance] removeDelegate:self];
-    return;
-  }
+- (void)bannerAdDidFailToLoad:(VungleBanner *)banner withError:(NSError *)error {
+  NSError *gadError = GADMAdapterVungleErrorToGADError(GADMAdapterVungleErrorAdNotPlayable,
+                                                       error.code, error.localizedDescription);
+  _adLoadCompletionHandler(nil, gadError);
 }
 
-- (void)adNotAvailable:(nonnull NSError *)error {
-  if (_isAdLoaded) {
-    // Already invoked an ad load callback.
-    return;
-  }
-  _adLoadCompletionHandler(nil, error);
+- (void)bannerAdWillPresent:(VungleBanner *)banner {
+  // No-op.
 }
 
-- (void)willShowAd {
-  self.bannerState = BannerRouterDelegateStatePlaying;
+- (void)bannerAdDidPresent:(VungleBanner *)banner {
```

Comment:
Will this callback be called when banner ad is presented? Or when the modal is presented after the banner ad is clicked?

Reply:
This callback will be called when the banner ad is presented. We do not have an equivalent callback for when the modal is presented after the banner ad is clicked.